x/ibc/02-client/keeper/keeper.go

```diff
@@ -197,6 +200,48 @@ func (k Keeper) GetSelfConsensusState(ctx sdk.Context, height uint64) (exported.
 	return consensusState, true
 }
 
+// ValidateSelfClient validates the client parameters for a client of the running chain
+// This function is only used to validate the client state the counterparty stores for this chain
+func (k Keeper) ValidateSelfClient(ctx sdk.Context, clientState exported.ClientState) error {
```

Comment:
This doesn't check:
- Max clock drift
- Frozen height

Maximum clock drift can vary, so I think it's alright not to check that, but we should just make sure there aren't any values which will cause Tendermint to throw errors.

We should probably verify that the latter is `0` initially? Not required security-wise though.

Once we add an epoch number we'll need to check that.

Reply:
Yes, I can add a frozen check, also epoch is implicitly checked by checking equality on `chainID`